pinot-core/src/test/java/org/apache/pinot/queries/JsonPathQueriesTest.java

```diff
@@ -55,17 +60,37 @@
   private static final String LONG_COLUMN = "longColumn";
   private static final String STRING_COLUMN = "stringColumn";
   private static final String JSON_COLUMN = "jsonColumn";
+  private static final String RAW_JSON_COLUMN = "rawJsonColumn";
+  private static final String RAW_BYTES_COLUMN = "rawBytesColumn";
+  private static final String DICTIONARY_BYTES_COLUMN = "dictionaryBytesColumn";
+  private static final String RAW_STRING_COLUMN = "rawStringColumn";
+  private static final String DICTIONARY_STRING_COLUMN = "dictionaryStringColumn";
   private static final String JSON_COLUMN_WITHOUT_INDEX = "jsonColumnWithoutIndex";
 
   private static final Schema SCHEMA = new Schema.SchemaBuilder().setSchemaName(RAW_TABLE_NAME)
       .addSingleValueDimension(INT_COLUMN, FieldSpec.DataType.INT)
       .addSingleValueDimension(LONG_COLUMN, FieldSpec.DataType.LONG)
       .addSingleValueDimension(STRING_COLUMN, FieldSpec.DataType.STRING)
       .addSingleValueDimension(JSON_COLUMN, FieldSpec.DataType.JSON)
+      .addSingleValueDimension(RAW_JSON_COLUMN, FieldSpec.DataType.JSON)
+      .addSingleValueDimension(RAW_BYTES_COLUMN, FieldSpec.DataType.BYTES)
```

Comment:
I added these because I think the test setup is very good, it can actually be used to test cases on `jsonextractscalar` that weren't previously tested, and these columns are only used to test `jsonextractscalar`. Indeed, JSON literals did not work and still do not work with `BYTES` and `STRING` Does this alleviate your concern?

Reply:
Unrelated to this PR. @amrishlal when did we make the decision that json_extract should work only on JSON datatype? this is a backward-incompatible change.

Reply:
@kishoreg I don't think that's what @amrishlal was saying, this test focuses on jsonpath literals like `select * from table where foo.bar = 0` and these only work on JSON columns, so adding these other columns here so `jsonextractscalar` can be tested using the thorough set up this tests provides doesn't feel appropriate.

Reply:
@richardstartin This unit test file is only to test JSON path expressions on `JSON` column. If you are trying to validate JSON_EXTRACT_SCALAR function, I think a better place for it would be JsonDataTypeQueriesTest.java which has a lot of test cases for json_extract_scalar. I haven't looked through the PR in detail, so maybe I am missing something, but will look into more detail.

@kishoreg Perhaps that was just my own personal impression at the time we were code reviewing JSON datatype. I suppose it should be ok to allow `JSON_EXTRACT_SCALAR` to work over `BYTE`, but then there is no guarantee that `BYTE` or `STRING` would contain JSON data (?), i.e any valid or invalid JSON data could be put into a `STRING` or `BYTE` column. If those byes or string are ingested into `JSON` datatype column, then we know for sure that it is valid JSON and can treat it as such during query processing, indexing, etc.

Reply:
@amrishlal I split the tests so this test only tests JSON path expressions and not the extract scalar function, but moved the setup into a base class so it can be reused. Does this resolve this issue?

Reply:
@richardstartin Yes, it resolves the issue. Thanks.